src/savegame_json.cpp

```diff
@@ -2948,6 +2948,51 @@ void mission::serialize( JsonOut &json ) const
     json.end_object();
 }
 
+void npc_work_log_entry::serialize( JsonOut &json ) const
+{
+    json.start_object();
+    json.member( "entry_npc_name", entry_npc_name );
+    json.member( "entry_timestamp", entry_timestamp );
+    json.member( "entry_npc_id", entry_npc_id );
+    json.member( "entry_type", static_cast<int>( entry_type ) );
+    json.member( "entry_description", entry_description );
+    json.member( "entry_position", entry_position );
+    if( entry_reason ) {
+        json.member( "reason_reason", static_cast<int>( entry_reason->reason ) );
+        json.member( "can_do", entry_reason->can_do );
+        json.member( "con_idx", entry_reason->con_idx );
+        json.member( "added_reason", entry_reason->added_reason );
+    } else {
+        json.member( "entry_reason", -1 );
```

Comment:
The JSON library supports direct serialization of `cata::optional`. You can add serialize/deserialize to `activity_reason_info` and write directly using `json.member( "entry_reason", entry_reason )`.